File: 148/SortList.scala
object SortList {
  class ListNode(val _x: Int = 0) {
    var x: Int = _x
    var next: ListNode = null
  }

  def sortList(head: ListNode): ListNode = {
    if (head == null) return null
    else {
      var curr = head
      var nodes = List.empty[ListNode]
      while (curr != null) {
        val node = new ListNode(curr.x)
        nodes = nodes :+ node
        curr = curr.next
      }

      nodes.sortBy(node => node.x)
      for (i <- 0 until nodes.length-1)
        nodes(i).next = nodes(i+1)
      nodes(0)
    }

  }

  def main(args: Array[String]): Unit = {
    val four = new ListNode(4)
    val two = new ListNode(2)
    val one = new ListNode(1)
    val three = new ListNode(3)
    one.next = three
    two.next = one
    four.next = two
    println(sortList(four).x)
  }

}

File: 148/sort-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) 
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var sortList = function (head) {
    if (!head)
        return null;
    let nodes = [];
    while (head) {
        let node = new ListNode(head.val);
        nodes.push(node);
        head = head.next;
    }

    nodes.sort((a, b) => a.val - b.val);
    for (let i = 0, len = nodes.length; i < len; i++) {
        nodes[i].next = nodes[i+1];
    }
    return nodes[0];
};